### PR TITLE
[FBX] general purpose fixes 🎄 

### LINKED ISSUE #43846: [FBX] Weird graphical artifacts in rigged mesh instance imported from Maya 2016 exported FBX file.
**Godot version:**
Tried with both Godot 3.2.3-stable and 3.2.4-beta2

**OS/device including version:**

Windows 10 Pro, NVIDIA RTX 3080, GLES3

**Issue description:**
When importing an FBX file with 2 meshes and a rig created in Maya 2016, the import succeeds with the following hierarchy:
hierarchy

In both the editor and game viewports, however, The following rendering issue of the mesh is observed:
weird_cull

I've ensured back-face culling is enabled on the material.
I've also observed the following:

1. If I clear the skin from the body MeshInstance, the issue persists
2. If I unassigned the skeleton from the body MeshInstance (regardless of whether skin resource is still assigned or not) the issue goes away.
3. If I import the FBX file into blender it seems fine. Same for Paint3D on Windows.
4. If I export the FBX file as a glTF from Blender and import into Godot, it works fine.
5. If I export the FBX file as another FBX file from Blender and import _that_ into Godot, the rig is entirely fubar, but I believe this is an entirely separate issue.

From all of these observations, I can conclude that there is something very peculiar happening when importing an FBX file created in Maya 2016 specifically (and possibly other versions of Maya as well--I don't have any other versions on hand to test).

Here is a log file of the verbose (-v) output when importing. No errors--everything seems to have imported fine:
[import.log](https://github.com/godotengine/godot/files/5595298/import.log)

**Steps to reproduce:**
I'm not sure if it's something unique to this mesh or not and I am unable to share the FBX file in question publicly. I can provide it privately to a Godot dev if needed for testing, however.

All I did was import said FBX file into Godot and view it in the editor viewport (though it also occurs when running a game).

**Minimal reproduction project:**
See above. Can't share FBX file publicly, but can provide privately if needed.

### PR DESCRIPTION
**Pull request by RevoluPowered**
### Improving FBX support
- Better materials
  - default values will no longer trigger things like emission, clear coat being enabled etc incorrectly, mostly importers misunderstand this value, a value of 1 is not actually enabled emission, it must have a non zero setting other than emission value for the emission flags to actually be enabled correctly in our engine #42386 
  - lambert materials are warned against significantly, do not use Lambert in Godot, use a PBR material like Ai Standard Surface (it's like going from low quality to high definition in terms of output in scenes and on assets, other engines just make you build the material from scratch)
  - roughness values are calculated correctly in this version
- Fixes for normal's array - some normal's from some files were dropped and generated, this is now fixed.
  - FBX indexing for items with (-1) index, for normal data is supported, this is super helpful at increasing our range of compatibility of FBX meshes.
- Fix bone rest data **no longer requiring go to bind pose in Maya and various applications**
  - Partial fix for #43688, animations & skinning needs rebuilt to support this correctly
- Validation tools added for validating large projects
- Provide package name, vendor and FBX vendor in the log.
- Implemented metadata properties so we can read extra document info required for bug reporting
- **FBX 2011 (version 7200)** is unsupported now, you must re-export your file in Maya or Max to upgrade the file format.
- Fixes skin bind poses being generated based on node xforms in the scene, this is fixed.
- Fixes duplicate bones being added to skin and prevents add_bones from registering skeleton bones it now registers skin bind poses, but this should really be documented in the engine correctly right now it doesn't tell you this is the case.

### known bugs:
- euler to quat is failing for some files
- blender is not exporting skins in the correct format.
~~- localise skeleton localises but loses important joint orient information (will write better algorithm at end for this to keep skeleton looking good in the engine)~~
~~- skinning is disabled & animations are broken by the new changes this will be fixed in a following commit.~~

*Bugsquad edit:* Fixes #43846.

### CODE REVIEW

**Comment by RevoluPowered** [comment]
OK so #43688 looks like a Euler To Quaternion bug I will dump them all to CSV and make a conversion utility.

File: drivers/gles3/rasterizer_storage_gles3.cpp [hunk 1]
```diff
@@ -4305,7 +4305,7 @@ AABB RasterizerStorageGLES3::mesh_get_aabb(RID p_mesh, RID p_skeleton) const {
 				const bool *skused = mesh->surfaces[i]->skeleton_bone_used.ptr();
 
 				int sbs = sk->size;
-				ERR_CONTINUE(bs > sbs);
+				ERR_CONTINUE_MSG(bs > sbs, "skeleton bone count doesn't match the skin bind count");
```
Comment (RevoluPowered) [inline drivers/gles3/rasterizer_storage_gles3.cpp:4308]: this should be removed it was only for debugging

**Comment by Riteo** [comment]
> blender is not exporting skins in the correct format.

Is this an issue with the importer or Blender itself? If the latter, is there a ticket open on their bug tracker?

**Comment by RevoluPowered** [comment]
> > blender is not exporting skins in the correct format.
> 
> Is this an issue with the importer or Blender itself? If the latter, is there a ticket open on their bug tracker?

The SDK is expecting the skin in global bind pose clusters for the mesh and they export in their own format. I'm going to be discussing this with them directly, but for now Blender FBX won't work for animated meshes. I've already said in the past their exporter can't export to the official SDK correctly skinned meshes, you can't import blender fbx into maya. It's one way at the moment because of this Maya to Blender but not back the way.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**